### PR TITLE
Attach event handlers dynamically in InstructorCourseDetails #5751

### LINKED ISSUE #5751: Attach event handlers dynamically in InstructorCourseDetails
The current approach uses html `onclick=""` to trigger a confirmation prompt and uses return false to prevent the link from working if user clicks no. This which works very well when using default browser dialog, but is not readily modifiable to use custom modals.

### PR DESCRIPTION
**Pull request by YongJieYongJie**
Fixes #5751

### CODE REVIEW

**Comment by chowyb** [comment]
Changes look good :+1:, although on a related note, should we use the course ID or course name for the popups? Your current implementation uses course IDs, although it is referred to as "course name" at some points, so we might have to change the names of the data attributes a little.

**Comment by YongJieYongJie** [comment]
@chowyb Renamed courseName to courseId!

**Comment by chowyb** [comment]
Good to go!

File: src/main/webapp/js/instructorCourseDetails.js [hunk 1]
```diff
@@ -44,27 +48,35 @@ function submitFormAjax() {
...
-/**
- * Functions to trigger registration key sending to a specific student in the
- * course.
- * Currently no confirmation dialog is shown.
- */
-function toggleSendRegistrationKey() {
-    return confirm('Usually, there is no need to use this feature because TEAMMATES '
-                   + 'sends an automatic invite to students at the opening time of each'
-                   + ' session. Send a join request anyway?');
+function attachEventToRemindStudentsButton() {
```
Comment (damithc) [inline src/main/webapp/js/instructorCourseDetails.js]: Try to write stateless functions. i.e. the function takes all it needs as parameters and returns the result. That way, the function becomes easier to test. The testability and test coverage of our current js code is not that good.

**Pull request merged**